client/src/lib/supabase/middleware.ts

```diff
@@ -2,46 +2,26 @@ import { createServerClient } from "@supabase/ssr";
 import { NextResponse, type NextRequest } from "next/server";
 import { userHasExistingProfile } from "@/lib/actions/profile_setup"
 
-console.log("***** BEGIN middleware.ts ******");
-console.log("middleware.ts is always invoked!!!");
-
-
 //looked into making these Sets but couldn't quite implement the references yet
 //allowed public paths
-const publicPaths = ["/","/login","/signup","/about","/auth/callback"];
+const publicPaths = new Set(["/","/login","/signup","/about","/auth/callback"]);
 
 //whitelists auth'd user paths
-const protectedPaths = ["/message", "/explore", "/profile", "/thread", "/track", "/profile_setup"];
+const protectedPaths = new Set(["/message", "/explore", "/profile", "/thread", "/track", "/profile_setup"]);
 
-//may be a JavaScript way to do this easier with Sets... looking into it
-function hasBasePath(pathname: string, basepaths: string[]) {
-  for (let i = 0; i < basepaths.length; i++) {
-    console.log("pathname: " + pathname);
-    console.log("basepaths[i]" + basepaths[i]);
-    if (pathname == "/")
-      return true 
-    if (basepaths[i] == "/") {continue}
-    if (pathname.indexOf(basepaths[i]) == 0) {
-      return true
-    }
-  }
-  return false
+function hasBasePath(pathname: string, basePaths: Set<string>): boolean {
+  const res = basePaths.has(pathname);
```

Comment:
Not sure if I mentioned this but this only works if the pathname does not have subroutes. For example this won't work properly if the pathname is "/message/company" or "/message/private".

Reply:
I added logic to handle the message paths